### ISSUE TITLE
`apply_filter` fails with "TypeError: object of type 'NoneType' has no len()"

### ISSUE THREAD

**Issue by kozo2** [cytoscape/py4cytoscape#56]
If you run https://colab.research.google.com/github/cytoscape/py4cytoscape/blob/0.0.10/doc/tutorials/Filtering-Networks.ipynb
you can reproduce the error below in the last cell.

```
    425         if len(sel_nodes) != 0:
    426             res= style_bypasses.hide_nodes(network_selection.invert_node_selection(network=network, base_url=base_url)['nodes'])
--> 427         if len(sel_edges) != 0:
    428             res = style_bypasses.hide_edges(network_selection.invert_edge_selection(network=network, base_url=base_url)['edges'])
    429 

TypeError: object of type 'NoneType' has no len()
```

**Comment by bdemchak** [comment]
This looks like a real bug involving doing operations on either selected nodes or selected edges when there are no selected nodes or edges. I can see a few places where this could blow up ... like in the filters example you showed, or in bypass properties (for example). 

The issue is that get_selected_nodes() and get_selected_edges() returns None instead of [] when there is nothing selected. This tracks the RCy3 behavior. The problem occurs when len() is evaluated on the result ... len(None) is undefined.

It doesn't look like RCy3 has a similar problem as it attempts to check "is.na(NA[1])", which does return a sensible result.

It'll take a few hours to track this down and write appropriate tests for this.

Thanks!

**Comment by bdemchak** [comment]
I think I have fixed this. It was a problem with a few facets.

It was caused by get_selected_edges() returning None because there were no selected edges. There are similar problems when there would be no selected nodes, or when calling a hide_selected_nodes() or hides_selected_edges() function.

The common problem appears when a function checks for selected edges/nodes and then does something with the resulting list. I added test suite checks for all of these situations.

A somewhat related problem occurs when passing None as an edge or node list to functions like set_node_label_opacity_bypass() ... there are lots of examples of this. The general advice here is to not pass None as a node/edge list. This is under the control of the caller, so I did not address this. Note that this is all by design, and isn't an error.

The bigger issue arises because this py4cytoscape code was created from the RCy3 code. While None (in Python) is analogous to RCy3 (in R), their runtime properties are pretty different. For example, in Python None[1] will throw an error while in R NA[1] will not. So, the consequences of RCy3 returning NA are different than py4cytoscape returning None ... and the result is that the caller needs to perform different checks on the py4cytoscape function results. This is simply the way things go with Python vs R. However, if we were designing these APIs now, we might want to avoid all reliance on None or NA, and use empty lists instead. We may never be in this position, though.
